code/datums/components/radioactive.dm

```diff
@@ -99,18 +99,19 @@
 	SIGNAL_HANDLER
 
 	if(QDELETED(src))
-		return
+		return NONE
 
 	if(!(clean_types & CLEAN_TYPE_RADIATION))
-		return
+		return NONE
 
 	if(!(clean_types & CLEAN_TYPE_WEAK))
 		qdel(src)
-		return
+		return NONE
```

Comment:
Stuff that delete source should probably return a value to stop the proc, no?

Reply:
Done